### PR TITLE
DSD-1052: Initial Footer component

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1052](https://jira.nypl.org/browse/DSD-1052)

## This PR does the following:

- Adds the initial version of the DS `Footer` component.
- This replicates the current production footer that exists on NYPL.
- Concerns that will be new tickets:
  - The DS social media icons don't match with production icons. There is already a ticket to get the new icons into the DS.
  - The media query breakpoints are different. Production uses these four: "320px", "768px", "1024px", "1500px". The DS uses these four: "320px", "600px", "960px", "1280px", "1536px". While not vastly different, the visual changes are noticeable. We should discuss this for a separate PR but do we want to replicate the legacy breakpoints and import them into the DS, or ignore them and use the current ones in the DS? 
- This is the React component. A solution still needs to be created for the static HTML/CSS version. This is TBD in a separate ticket.

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/Footer/Footer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    expect(icons[1]).toHaveAttribute("title", "NYPL on Twitter");
+    expect(icons[2]).toHaveAttribute("title", "NYPL on Instagram");
+    expect(icons[3]).toHaveAttribute("title", "NYPL on Youtube");
+  });
+
+  it("renders a footer image of the NYPL building", () => {
+    const logo = screen.getAllByRole("img")[4];
+
+    expect(logo).toBeInTheDocument();
+    expect(logo).toHaveAttribute(
```
Comment (bigfishdesign13) [inline src/components/Footer/Footer.test.tsx]: Why did you do this test by `src` and the logo by `title`?  Could you instead test for the `alt` text here?

File: src/theme/components/fieldset.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ interface FieldSetProps {
 
 const Fieldset = {
   baseStyle: ({ isLegendHidden }: FieldSetProps) => {
-    const screenreaderStyles = isLegendHidden ? screenreaderOnly : {};
+    const screenreaderStyles = isLegendHidden ? screenreaderOnly() : {};
```
Comment (bigfishdesign13) [inline src/theme/components/fieldset.ts:10]: Why this change?

File: src/theme/components/footer.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
+import { displayScreenreaderOnly, screenreaderOnly } from "./globalMixins";
+
+export const footerGray = "#54514A";
```
Comment (bigfishdesign13) [inline src/theme/components/footer.ts:3]: Great.  This gray sits almost half way between `ui.gray.dark` and `ui.gray.x-dark`.  If it was closer to one or the other, I would push to make the jump to our gray, but not when it's this far off.  Too bad.

File: src/theme/components/footer.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      fontWeight: "400",
+      margin: "0 0.2rem",
+      textDecoration: "none",
+      _hover: {
+        color: "ui.white",
+        textDecoration: "underline",
+      },
+      _focus: {
+        backgroundColor: footerGray,
+        outline: "solid 2px var(--nypl-colors-ui-white)",
```
Comment (bigfishdesign13) [inline src/theme/components/footer.ts]: For border styles, we've split this into two attributes so we could use the theme object: `border` and `borderColor`.  Maybe use the same approach here?

File: src/theme/components/globalMixins.ts [hunk 5]
```diff
@@ -2,13 +2,28 @@
...
  */
-const screenreaderOnly = {
+
+/**
+ * If the element that should be visually hidden should only be hidden on
+ * small screens, then pass in true for the `isMobileOnly` argument. Otherwise,
+ * this will set an element to always be visually hidden regardless of the
+ * screen size.
+ */
+const screenreaderOnly = (isMobileOnly = false) => ({
```
Comment (bigfishdesign13) [inline src/theme/components/globalMixins.ts:12]: Ignore my other question about this.  I see now why you made that other change.

**Review by bigfishdesign13 — COMMENTED** [review]
Overall, this look quite good.  I have a few questions sprinnkled throughout.

One thing that I noticed as I was comparing with the footer on NYPL.org is that we use font-smoothing, whereas they do not.  Before we roll out the `Header` and `Footer` components, we may need to make a push to have that attribute added to other projects.  It would be weird to have a page with mixed font treatments.

Why are the breakpoints so different between Reservoir and DXP?  Are the breakpoints you are referring to the same as what Bill is using in Scout?  If so, how does he deal with the discrepancies?

**Comment by EdwinGuzman** [comment]
> One thing that I noticed as I was comparing with the footer on NYPL.org is that we use font-smoothing, whereas they do not. Before we roll out the Header and Footer components, we may need to make a push to have that attribute added to other projects. It would be weird to have a page with mixed font treatments.

I don't have a keen eye to notice this right away... can you make a ticket for this?

> Why are the breakpoints so different between Reservoir and DXP? Are the breakpoints you are referring to the same as what Bill is using in Scout? If so, how does he deal with the discrepancies?

The breakpoints from the Header/Footer are from the older design system based on what was at that point, the most common breakpoints for all devices. These have change since now devices have vastly different sizes. It looks like Scout is using the DS breakpoints: https://github.com/NYPL/dxp-react-search/blob/development/src/styles/base/_breakpoints.scss

Instead of importing the Header/Footer breakpoints, we are building the two new components with the DS breakpoints. That's the only big discrepancy between the old and new (besides the font treatment).

File: src/components/Footer/Footer.test.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Footer/Footer.test.tsx]: It didn't work at first but I think it'll work with the latest updates from `development`. I'll try it and see if it works, otherwise, the reason I did the different tests is because the DOM didn't pick it up.

**Comment by EdwinGuzman** [comment]
Updated. Got the image by alt text now in the unit test.

**Review by bigfishdesign13 — APPROVED** [review]
All good.  However, it looks like there are a lot more modified files now.  The merge from development must have done that?

**Comment by EdwinGuzman** [comment]
I had to update the base branch from `development`, now it's down to 11 changed files.
